src/app/(authenticated)/(dashboard)/[publicId]/settings/api/components/table.tsx

```diff
@@ -103,17 +137,23 @@ const ApiKeysTable = ({ keys }: { keys: ApiKey[] }) => {
                         Rotate key
                       </DropdownMenuItem>
 
-                      <ConfirmDialog
```

Comment:
Any reason why this was replaced by `Alert`, this is supposed to be a re-usable component for the same purpose?

Reply:
Sometimes reusable components like `AlertDialog` won't work when nested inside components like dropdowns and may have some strange bugs due to event propagation. The bug I encountered is that whenever the alert is opened, it automatically closes.

The reason the previous code worked is because it just used a `<div>` as a trigger and did not use a dropdown item component. This approach is not accessible and defeats the purpose of HTML semantics. Sometimes it's easy to use the compound component rather than adding prop for each specific use case for the reusable component.